cmd/metal-api/internal/metal/errors.go

```diff
@@ -20,7 +18,7 @@ func NotFound(format string, args ...interface{}) error {
 
 // IsNotFound checks if an error is a notfound error.
 func IsNotFound(e error) bool {
-	return errors.Cause(e) == errNotFound
+	return errors.Is(errors.Cause(e), errNotFound)
```

Comment:
are these functions covered by a  test ? if yes this is great

Reply:
Done.